python/hail/contrib/ldsc.py

```diff
@@ -0,0 +1,156 @@
+
+import numpy as np
+import hail as hl
+from hail.table import Table
+from hail.linalg import BlockMatrix
+from hail.typecheck import *
+from hail.expr.expressions import *
+from hail.utils import new_temp_file, wrap_to_list
+
+@typecheck(entry_expr=expr_numeric,
+           annotation_exprs=oneof(expr_numeric, sequenceof(expr_numeric)),
+           position_expr=expr_numeric,
+           window_size=oneof(int, float))
+def ld_score(entry_expr, annotation_exprs, position_expr, window_size) -> Table:
+    """
+    Calculate LD scores.
+
+    Example
+    -------
+
+    >>> # Load genetic data into MatrixTable
+    >>> mt = hl.import_plink(bed='data/ldsc.bed', 
+    ...                      bim='data/ldsc.bim', 
+    ...                      fam='data/ldsc.fam')
+    
+    >>> # Create locus-keyed Table with numeric variant annotations
+    >>> ht = hl.import_table('data/ldsc.annot', types={'BP': hl.tint, 'binary': hl.tfloat, 'continuous': hl.tfloat})
+    >>> ht = ht.annotate(locus=hl.locus(ht.CHR, ht.BP))  
+    >>> ht = ht.key_by('locus')
+    
+    >>> # Annotate MatrixTable with external annotations 
+    >>> mt = mt.annotate_rows(univariate_annotation=hl.literal(1),   # univariate LD scores
+    ...                       binary_annotation=ht[mt.locus].binary, 
+    ...                       continuous_annotation=ht[mt.locus].continuous)
+    ...
+    >>> # Annotate MatrixTable with alt allele count stats
+    >>> mt = mt.annotate_rows(stats=hl.agg.stats(mt.GT.n_alt_alleles()))
+     
+    >>> # Calculate LD score for each variant, annotation using standardized genotypes
+    >>> ht_scores = hl.ld_score(entry_expr=hl.or_else((mt.GT.n_alt_alleles() - mt.stats.mean)/mt.stats.stdev, 0.0),
+    ...                         annotation_exprs=[mt.univariate_annotation, mt.binary_annotation, mt.continuous_annotation],
+    ...                         position_expr=mt.cm_position,
+    ...                         window_size=1)
+
+    Warning
+    -------
+    :func:`.ld_score` will fail if ``entry_expr`` results in any missing values. The special float value ``nan`` is not considered a missing value.
+
```

Comment:
unnecessary extra space

Reply:
Is this referencing the double newline after the warning section? Without the extra newline, the "Further Reading" section is lumped into the warning section.

Reply:
Ah. The best thing to do then is probably tab over the line after the warning. Numpy docstrings have some weird rules.